File: android/src/main/java/com/getcapacitor/community/classes/results/IsEstablishedResult.java
package com.getcapacitor.community.classes.results;

import com.getcapacitor.JSObject;
import com.getcapacitor.community.interfaces.Result;

public class IsEstablishedResult implements Result {

    private final Boolean isEstablished;

    public IsEstablishedResult(Boolean isEstablished) {
        this.isEstablished = isEstablished;
    }

    @Override
    public JSObject toJSObject() {
        JSObject result = new JSObject();

        if (isEstablished != null) result.put("isEstablished", isEstablished);

        return result;
    }
}

File: ios/Sources/BitchatPlugin/Classes/Results/IsEstablishedResult.swift
import Foundation
import Capacitor

@objc public class IsEstablishedResult: NSObject, Result {
    let isEstablished: Bool?

    init(_ isEstablished: Bool?) {
        self.isEstablished = isEstablished
    }

    public func toJSObject() -> AnyObject {
        var result = JSObject()

        if let isEstablished = isEstablished {
            result["isEstablished"] = isEstablished
        }

        return result as AnyObject
    }
}
